fsspec/implementations/tests/test_local.py

```diff
@@ -823,3 +823,36 @@ def test_put_file_to_dir(tmpdir):
     fs.put(src_file, target_dir)
 
     assert fs.isfile(target_file)
+
+
+def test_du(tmpdir):
+    file = os.path.join(tmpdir, "file")
+    subdir = os.path.join(tmpdir, "subdir")
+    subfile = os.path.join(subdir, "subfile")
+
+    fs = LocalFileSystem()
+    with open(file, "wb") as f:
+        f.write(b"4444")
+    fs.mkdir(subdir)
+    with open(subfile, "wb") as f:
+        f.write(b"7777777")
+
+    assert fs.du(tmpdir) == 11
+    assert fs.du(tmpdir, total=False) == {file: 4, subfile: 7}
+    # Note directory size is OS-specific, but must be positive
+    assert fs.du(tmpdir, withdirs=True) > 11
+
+    d = fs.du(tmpdir, total=False, withdirs=True)
+    assert len(d) == 4
+    assert d[file] == 4
+    assert d[subfile] == 7
+    assert d[tmpdir] > 0
+    assert d[subdir] > 0
+
+    assert fs.du(tmpdir, maxdepth=2) == 11
+    assert fs.du(tmpdir, maxdepth=1) == 4
+    assert fs.du(tmpdir, maxdepth=0) == 4
```

Comment:
It surprised me initially that `maxdepth=0` gives the same output as `maxdepth=1`. Maybe it does makes sense though, `maxdepth` effectively has a minimum value of 1.

Reply:
Is maxdeph <=0 a ValueError then?

Reply:
No, currently `maxdepth <= 1` proceeds as though `maxdepth=1`.

Linux command-line `du` is maybe relevant here. A `maxdepth` less than `0` gives an error, but `maxdepth=0` when specifying a directory path gives you just the directory back without any recursion. So the equivalent behaviour in `fsspec` would be for line 854 above
```python
assert fs.du(tmpdir, maxdepth=0) == 0
```

We could classify this as a bug which should raise an error on `maxdepth<0` and handle `maxdepth=0` differently.

Reply:
OK, not going to worry about the edge case here. Anyone trying this is just being silly :)